src/west/app/project.py

```diff
@@ -1602,6 +1602,204 @@ def do_run(self, args, user_args):
                 failed.append(project)
         self._handle_failed(args, failed)
 
+GREP_EPILOG = '''
+EXAMPLES
+--------
+
+To get "git grep foo" results from all cloned, active projects:
+
+  west grep foo
+
+To do the same:
+
+- with "git grep --untracked": west grep --untracked foo
+- with "ripgrep":              west grep --tool ripgrep foo
+- with "grep --recursive":     west grep --tool grep foo
+
+To switch the default tool:
+
+- to ripgrep:  west config grep.tool ripgrep
+- to grep:     west config grep.tool grep
+
+GREP TOOLS
+----------
+
+This command runs a "grep tool" command in the top directory of each project.
+Supported tools:
+
+- git-grep (default)
+- ripgrep
+- grep
+
+Set the "grep.tool" configuration option to change the default.
+
+Use "--tool" to switch the tool from its default.
+
+TOOL PATH
+---------
+
+Use --tool-path to override the path to the tool. For example:
+
+  west grep --tool ripgrep --tool-path /my/special/ripgrep
+
+Without --tool-path, the "grep.<TOOL>-path" configuration option
+is checked next. For example, to set the default path to ripgrep:
+
+  west config grep.ripgrep-path /my/special/ripgrep
+
+If the option is not set, "west grep" searches for the tool as follows:
+
+- git-grep: search for "git" (and run as "git grep")
+- ripgrep: search for "rg", then "ripgrep"
+- grep: search for "grep"
+
+TOOL ARGUMENTS
+--------------
+
+The "grep.<TOOL>-args" configuration options, if set, contain arguments
+that are always passed to the tool. The defaults for these are:
+
+- git-grep: (none)
+- ripgrep: (none)
+- grep: "--recursive"
+
+Additional command line arguments not recognized by "west grep" are
+passed to the tool after that. For example:
+
+  west grep --this-option-is-passed-to-the-tool
+
+To force an argument to be given to the tool instead of west, put
+it after a "--", like this:
+
+  west grep -- --project=val
```

Comment:
In theory it should be possible to use `--` twice, correct? The first one for `west`, the second `--` for `grep`. I think having two `--` in one test case would be great for test coverage.